bindings/wasm/src/did/wasm_core_document.rs

```diff
@@ -513,11 +583,45 @@ struct ICoreDocumentHelper {
   properties: Object,
 }
 
-impl_wasm_json!(WasmCoreDocument, CoreDocument);
-impl_wasm_clone!(WasmCoreDocument, CoreDocument);
+#[wasm_bindgen(js_class = CoreDocument)]
+impl WasmCoreDocument {
+  /// Serializes this to a JSON object.
+  #[wasm_bindgen(js_name = toJSON)]
+  pub fn to_json(&self) -> crate::error::Result<JsValue> {
+    use crate::error::WasmResult;
+    JsValue::from_serde(&self.0.blocking_read().as_ref()).wasm_result()
+  }
+
+  /// Deserializes an instance from a JSON object.
+  #[wasm_bindgen(js_name = fromJSON)]
+  pub fn from_json(json: &JsValue) -> crate::error::Result<WasmCoreDocument> {
+    use crate::error::WasmResult;
+    json
+      .into_serde()
+      .map(|value| Self(Rc::new(CoreDocumentLock::new(value))))
+      .wasm_result()
+  }
+}
```

Comment:
Can you add this to the impl block above and use the conventional `WasmResult` import?